### PR TITLE
Download Azure billing CSV to filesystem, instead of holding in memory

### PR DESCRIPTION
**Pull request by thomasvn**
## What does this PR change?
* Downloads the Azure billing CSV to the filesystem, instead of holding it in memory

## Does this PR relate to any other PRs?
* No

## How will this PR impact users?
* For users with large Azure bills, this configuration should consume less memory.

## Does this PR address any GitHub or Zendesk issues?
* None

## How was this PR tested?

```sh
export CONFIG_PATH="/tmp/localrun/default"
export PROMETHEUS_SERVER_ENDPOINT="http://127.0.0.1:9080"
export CLOUD_COST_ENABLED="true"
mkdir -p $CONFIG_PATH/cloud-integration/
cp ~/cloud-integration.json $CONFIG_PATH/cloud-integration/cloud-integration.json

# Validated CloudCost ran as expected.
# Validated that Azure billing CSV files got downloaded to /tmp/localrun/default/db/cloudcost.
go run cmd/costmodel/main.go
```

## Does this PR require changes to documentation?
* Not yet.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

File: pkg/cloud/azure/storagebillingparser.go [hunk 1]
```diff
@@ -56,17 +60,40 @@ func (asbp *AzureStorageBillingParser) ParseBillingData(start, end time.Time, re
...
-		blobBytes, err2 := asbp.DownloadBlob(blobName, client, ctx)
-		if err2 != nil {
-			asbp.ConnectionStatus = cloud.FailedConnection
-			return err2
-		}
-		err2 = asbp.parseCSV(start, end, csv.NewReader(bytes.NewReader(blobBytes)), resultFn)
-		if err2 != nil {
-			asbp.ConnectionStatus = cloud.ParseError
-			return err2
+		if env.IsAzureDownloadBillingDataToDisk() {
```
Comment (Sean-Holcomb) [inline pkg/cloud/azure/storagebillingparser.go]: I don't see the need for this if here, if this is truly is an improvement there is no need for YA env variable. Lets just always do it like this.

File: pkg/cloud/azure/storagebillingparser.go [hunk 2]
```diff
@@ -184,3 +211,38 @@ func (asbp *AzureStorageBillingParser) timeToMonthString(input time.Time) string
 	endOfMonth := input.AddDate(0, 1, -input.Day())
 	return startOfMonth.Format(format) + "-" + endOfMonth.Format(format)
 }
+
+// deleteFilesOlderThan7d recursively walks the directory specified and deletes
+// files which have not been modified in the last 7 days. Returns a list of
+// files deleted.
+func (asbp *AzureStorageBillingParser) deleteFilesOlderThan7d(localPath string) ([]string, error) {
```
Comment (Sean-Holcomb) [inline pkg/cloud/azure/storagebillingparser.go:207]: This doesn't quite feel right to me, but I think it is fine for now. If we do end up supporting partitioned exports doing it this way will make thing easier.

**Review by Sean-Holcomb — APPROVED** [review]
I think it is best to get rid of the env variable and download the file for all users. Memory improvements should benefit everyone here

File: pkg/cloud/azure/storagebillingparser.go (hunk 2 above)
Comment (thomasvn) [reply pkg/cloud/azure/storagebillingparser.go:207]: Agree. Deleting files based off an arbitrary time cutoff is weird.

The goal though, was to make sure that local disk space doesn't fill up too quickly. With CloudCost running, we would eventually be downloading every Azure month-to-date bill to disk. We should clean up the files that we're likely no longer using?

File: pkg/cloud/azure/storageconnection.go [hunk 3]
```diff
@@ -45,24 +46,32 @@ func (sc *StorageConnection) getBlobURLTemplate() string {
 	return "https://%s.blob.core.windows.net/%s"
 }
 
-func (sc *StorageConnection) DownloadBlob(blobName string, client *azblob.Client, ctx context.Context) ([]byte, error) {
```
Comment (thomasvn) [inline pkg/cloud/azure/storageconnection.go:48]: @Sean-Holcomb I found no other references to `DownloadBlob()` so I've deleted it. Let me know if we need to bring it back.

File: pkg/cloud/azure/storagebillingparser.go (hunk 1 above)
Comment (thomasvn) [reply pkg/cloud/azure/storagebillingparser.go]: Made the changes! This is now the default functionality.

**Pull request merged**